slides/week3/git.org

```diff
@@ -0,0 +1,200 @@
+    git add -A
+    git commit -m "Add stable_release.txt"
+  #+END_SRC
+- This will *not affect* any other branches
+- The other branches will stay behind on 'initial commit'
+  #+BEGIN_SRC
+    5592388 * stable      Add stable_release.txt
+    844a162 * master beta Initial commit
+  #+END_SRC
+
+** First Beta Commit
+- Let's make a experimental commit on the beta branch
+  #+BEGIN_SRC sh
+    git checkout beta
+    echo "This feature is unstable!" > beta_release.txt
+    git add -A
+    git commit -m "Add beta_release.txt"
+  #+END_SRC
+- Since the beta branch was *behind* the stable branch, this creates a 'fork' in the commit history
+- Notice, master has not moved at all (since no commits have been made on it)
+  #+BEGIN_SRC
+    1c65f7b * beta Add beta_release.txt
+    5592388 | * stable Add stable_release.txt
+            |/
+    844a162 * master Initial commit
+  #+END_SRC
+- Play around, and checkout to the various branches!
+  #+BEGIN_SRC shell
+    git checkout <BRANCH>
+    ls
+  #+END_SRC
+
+** Let's make beta official
+- Let's bring the hard work from the beta branch onto the stable branch
+- Right now, the stable and the beta branch have 'forked', we need to bring them back together
+- This is a *merge*
+  #+BEGIN_SRC sh
+    # Checkout to the branch we want to merge **into** aka the "base branch"
+    git checkout stable
+    # Merge the branch we want (beta) into the current branch (stable)
+    git merge beta
+    # This will launch an editor, save and quit it to complete
+  #+END_SRC
+- This does not change any other branch
+  #+BEGIN_SRC
+    ddc2df9 *   stable Merge branch 'beta' into stable
+            |\
+    1c65f7b | * beta Add beta_release.txt
+    5592388 * | Add stable_release.txt
+            |/
+    844a162 * master Initial commit
+  #+END_SRC
+
+** Remotes vs Branches vs Forks
+- A fork or close does a full copy of the git repo
+- A remote is a pointer to another copy of this git repo
+- A repository is a collection of *branches*
+- You can only directly modify local branches, but:
+  - You can merge from a remote into your branch (pull)
+  - You can merge from your branch to a remote (push)
+
+* Pull requests
+- A pull request = PR
+- A request for a project owner to merge from your fork to their copy
+  #+BEGIN_SRC fundamental
+          push     PR
+    laptop -> fork -> upstream
+      ^         |        |
+      |         v        v
+      |-------------------
+               pull
+  #+END_SRC
+- Request an owner to merge *from a branch on your fork*
+
+** Make a pull request
+- Push code locally stored on your computer to your fork on GitHub
+  #+BEGIN_SRC shell
+    git push origin stable
+  #+END_SRC
+- Locate your fork on GitHub and click "New pull request"
+[[file:https://i.imgur.com/8xwEajp.jpg]]
+
+** Make a pull request (continued)
+- Ensure that both repos and branches are correct
+- Click "Create pull request"
+[[file:https://i.imgur.com/gxUa2Zx.jpg]]
```

Comment:
Per Slack we decided to demonstrate a pull request without everyone actually submitting one. I'd replace this slide with a simple demo title slide.

Still need to work out where that's going to happen exactly...